actionpack/CHANGELOG.md

```diff
@@ -1,3 +1,29 @@
+*  Cookies `:expires` option supports `ActiveSupport::Duration`
+
+    Earlier, we had to specify the cookie expiry using a time object.
+    But now, you can now specify when the cookie expires using an `ActiveSupport::Duration`
+    object as well.
+
+    For example,
+
+    cookies[:user_name] = { value: "assain", expires: 1.hour }
+    cookies[:oreo] = { value: "chocolate cookies", expires: 6.months }
+
+    Pull Request: #30121
+
+    *Assain Jaleel*
+
+*  Cookie expiry is enforced for signed/encrypted cookies
+
+    By tacking on expiry meta data to the cookie value and verifying them on receipt,
```

Comment:
Let's reverse the 2 paragraphs to start with the security boost.

```
Rails can thwart attacks by malicious clients that don't honor a cookie's expiry.

It does so by stashing the expiry within the written cookie and relying on the
signing/encrypting to vouch that it hasn't been tampered with. Then on a
server-side read, the expiry is verified and any expired cookie is discarded.
```

This shit is a challenge to explain clearly in few words! So sorry for throwing your bit, but I really want us to be clear about this 😄

Reply:
Nice! :smile: